### ISSUE TITLE
Is there a rule for preventing the use of nested props without shape validation?

### ISSUE THREAD

**Issue by InsidersByte** [yannickcr/eslint-plugin-react#587]
This produces no errors.

```javascript
static propTypes = {
  item: React.PropTypes.object.isRequired,
}

return (
  <tr>
    <th>{this.props.item.name}</th>
    <th>{this.props.item.price}</th>
  </tr>
);
```

This results in an error.

* `error  'item.price' is missing in props validation      react/prop-types`

```javascript
static propTypes = {
  item: React.PropTypes.shape({
    name: React.PropTypes.string.isRequired,
   }).isRequired,
}

return (
  <tr>
    <th>{this.props.item.name}</th>
    <th>{this.props.item.price}</th>
  </tr>
);
```

I would like the first one to give an error stating something like nested prop-types need validation.

**Comment by ljharb** [comment]
I believe this is because the `item` propType is a function, and functions have a `name` property. @yannickcr, could this be the cause of the bug?

**Comment by novascreen** [comment]
Seems to me like in the first case you basically state that it just has to be an object with any shape, so you get no error.
In the second case you gave it a specific shape, but then you use a property that's not in your shape, so you get an error.
I don't think this is a bug, you'd basically have to disallow the `object` type if i'm not mistaken.

**Comment by yannickcr** [comment]
@novascreen You're right. We do not validate properties when the prop-type is `object` because it is too hazardous to try to differentiate native `object` properties and nested props. So every `this.props.item.xxx` are accepted.

We only do nested prop-type validation on `shape`, `arrayOf`, `objectOf` and `oneOfType`.

@InsidersByte To prevent the use of nested props without shape validation I suggest you to use the [forbid-prop-types](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-prop-types.md) rule to forbid the use of `React.PropTypes.object`.
